app/controllers/save_pages_controller.rb

```diff
@@ -0,0 +1,52 @@
+class SavePagesController < ApplicationController
+  include AccountConcern
+
+  before_action :set_no_cache_headers
+  before_action { head :not_found unless feature_flag_enabled? }
+
+  def create
+    GdsApi.account_api.save_page(
+      page_path: page_path,
+      govuk_account_session: account_session_header,
+    )
+    redirect_to page_path + "?personalisation=page_saved"
+  rescue GdsApi::HTTPUnauthorized
+    authenticate(save_page_path(page_path: page_path))
+  rescue GdsApi::HTTPUnprocessableEntity
+    redirect_to page_path + "?personalisation=page_not_saved"
+  end
+
+  def destroy
+    GdsApi.account_api.delete_saved_page(
+      page_path: page_path,
+      govuk_account_session: account_session_header,
+    )
+
+    redirect_to page_path + "?personalisation=page_removed"
+  rescue GdsApi::HTTPUnauthorized
+    authenticate(remove_saved_page_path(page_path: page_path))
+  end
+
+private
+
+  def page_path
+    ensure_is_path(params[:page_path])
+  end
+
+  def authenticate(redirect_path)
+    @account_session_header = nil
```

Comment:
Use `logout!` here, just setting the `@account_session_header` to `nil` won't set the end-session response header